src/key_io.cpp

```diff
@@ -113,18 +125,32 @@ CTxDestination DecodeDestination(const std::string& str, const CChainParams& par
                         return scriptid;
                     }
                 }
+
+                error_str = "Invalid Bech32 v0 address data size";
                 return CNoDestination();
             }
-            if (version > 16 || data.size() < 2 || data.size() > 40) {
+
+            if (version > 16) {
+                error_str = "Invalid Bech32 address version";
```

Comment:
suggested: "Invalid Bech32 address witness version" to clarify what thing's version is wrong